---
title: "NA NCR Hamming Distances Updated"
author: "Alexandra Christopoulos"
date: "2023-04-05"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(root.dir = "C:/Users/Lexie Christopoulos/Repos/GenomeSequencing-/db/",echo = TRUE)
```


```{r, include=FALSE}
#in console: setwd("C:/Users/Lexie Christopoulos/Repos/GenomeSequencing-/db/NA")
```


## NA H1N1 Forward 1.3 primer NCR hamming distances 

```{r}
NA_H1_F1 <- read.csv("hamming_distance_results/NA_H1_NA_Forward_1.3_results_NCR.csv")

NA_H1_1 <- NA_H1_F1[-c(48, 49),]

barplot(table(NA_H1_1$hamming_distance), main = "NA H1N1 hamming distances, forward 1.3 primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,50))
```

**NA Forward 1.3 primer**: *AGC ATC* **AGC AAA AGC AGG** AGT

**Melting temperature**: 57.3ºC 

**Hairpins** (ΔG < -1): 1 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 1 (with both reverse primers)

This primer is specific for the NA segment of H1N1, H3N2, and H9N2 strains of Influenza A. The primer includes a 6 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGC AAA AGC AGG, **bolded**),and part of the non-coding region that is unique to the NA segments. Based on the alignment shade threshold and hamming distance calculations, 97% of NA H1N1 sequences will have a hamming distance of 0 with this primer. 47 H1N1 sequences were used for this alignment and calculations. 


## NA H1N1 Reverse 2.2 primer NCR hamming distances


```{r}
NA_H1_R <- read.csv("hamming_distance_results/NA_H1_NA_Reverse_2.2_results_NCR.csv")

NA_H1_2 <- NA_H1_R[-c(48, 49),]

barplot(table(NA_H1_2$hamming_distance), main = "NA H1N1 hamming distances, reverse 2.2 primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,50))
 
```

**NA Reverse 2.2 primer**: *CGA TGC C***AG TAG AAA CAA GG**A GTT 

**Reverse compliment (without 5' tail)**: AAC T**CC TTG TTT CTA CT**

**Melting temperature**: 57.1ºC 

**Hairpins** (ΔG < -1): 0 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 1 (with all forward primers)

This primer is specific for the NA segment of H1N1, H3N2, and H9N2 strains of Influenza A. The reverse compliment of this primer was used for the alignments and hamming distance calculations. The primer includes a 7 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGT AGA AAC AAG G, **bolded**) and part of the non-coding region that is unique to the NA segments. Based on the alignment shade threshold and hamming distance calculations, 75% of NA H1N1 sequences will have a hamming distance of 0 with this primer. 47 H1N1 sequences were used for this alignment and calculations. 

**Note**: There are a few NA H1N1 sequences (the ones that have a hamming distance of 3) that are more complimentary to the NA Reverse 2.3 primer. I believe that a 4 primer pool for the NA primers would be the way to go, therefore this should not be an issue since this primer will also be present. 


## NA H3N2 Forward 1.3 primer NCR hamming distances 

```{r}
NA_H3_F1 <- read.csv("hamming_distance_results/NA_H3_NA_Forward_1.3_results_NCR.csv")

NA_H3_1 <- NA_H3_F1[-c(4677, 4678),]

barplot(table(NA_H3_1$hamming_distance), main = "NA H3N2 hamming distances, forward 1.3 primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,5000))

```


**NA Forward 1.3 primer**: *AGC ATC* **AGC AAA AGC AGG** AGT

**Melting temperature**: 57.3ºC 

**Hairpins** (ΔG < -1): 1 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 1 (with both reverse primers)

This primer is specific for the NA segment of H1N1, H3N2, and H9N2 strains of Influenza A. The primer includes a 6 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGC AAA AGC AGG, **bolded**),and part of the non-coding region that is unique to the NA segments. Based on the alignment shade threshold and hamming distance calculations, 90% of NA H3N2 sequences will have a hamming distance of 0 with this primer. 4,676 H3N2 sequences were used for this alignment and calculations. 


## NA H3N2 Reverse 2.2 primer NCR hamming distances 

```{r}
NA_H3_R <- read.csv("hamming_distance_results/NA_H3_NA_Reverse_2.2_results_NCR.csv")

NA_H3_2 <- NA_H3_R[-c(4677, 4678),]

barplot(table(NA_H3_2$hamming_distance), main = "NA H3N2 hamming distances, reverse 2.2 primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,5000))

```


**NA Reverse 2.2 primer**: *CGA TGC C***AG TAG AAA CAA GG**A GTT 

**Reverse compliment (without 5' tail)**: AAC T**CC TTG TTT CTA CT**

**Melting temperature**: 57.1ºC 

**Hairpins** (ΔG < -1): 0 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 1 (with all forward primers)

This primer is specific for the NA segment of H1N1, H3N2, and H9N2 strains of Influenza A. The reverse compliment of this primer was used for the alignments and hamming distance calculations. The primer includes a 7 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGT AGA AAC AAG G, **bolded**) and part of the non-coding region that is unique to the NA segments. Based on the alignment shade threshold and hamming distance calculations, 85% of NA H3N2 sequences will have a hamming distance of 0 with this primer. 4,676 H3N2 sequences were used for this alignment and calculations. 

## NA H5N6 Forward 1.4 primer NCR hamming distances 

```{r}
NA_H5_F <- read.csv("hamming_distance_results/NA_H5_NA_Forward_1.4_results_NCR.csv")

NA_H5_1 <- NA_H5_F[-c(13, 14),]

barplot(table(NA_H5_1$hamming_distance), main = "NA H5N6 Hamming Distances, forward 1.4 primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,15))
```


**NA Forward 1.4 primer**: *AGC ATC* **AGC AAA AGC AGG** GT

**Melting temperature**: 57.7ºC 

**Hairpins** (ΔG < -1): 1 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 1 (with both reverse primers)


This primer is specific for the NA segment of H5N1 and H7N9 strains of Influenza A. The primer includes a 6 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGC AAA AGC AGG, **bolded**),and part of the non-coding region that is unique to the NA segments. Based on the alignment shade threshold and hamming distance calculations, 92% of NA H5N6 sequences will have a hamming distance of 0 with this primer. 12 H5N6 sequences were used for this alignment and calculations.


## NA H5N6 Reverse 2.3 primer NCR hamming distances 

```{r}
NA_H5_R <-read.csv("hamming_distance_results/NA_H5_NA_Reverse_2.3_results_NCR.csv")

NA_H5_2 <- NA_H5_R[-c(13, 14),]

barplot(table(NA_H5_2$hamming_distance), main = "NA H5N6 Hamming Distances, reverse 2.3 primer",  xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,15))
```

**NA Reverse 2.2 primer**: *GAT GCC* **AGT AGA AAC AAG G**GT *S*T
 

**Reverse compliment (without 5' tail)**: A*C*A C**CC TTG TTT CTA CT**

**Melting temperature**: 57ºC 

**Hairpins** (ΔG < -1): 0 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 1 (with all forward primers)

This primer is specific for the NA segment of H5N6, H7N9, and some H1N1 strains of Influenza A. The reverse compliment of this primer was used for the alignments and hamming distance calculations. This is a mixed base primer, with the mixed base noted in *italics*. The nucleotide that is present in the H5N6 strains is noted in the reverse compliment sequence, also in *italics*. The primer includes a 6 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGT AGA AAC AAG G, **bolded**) and part of the non-coding region that is unique to the NA segments. Based on the alignment shade threshold and hamming distance calculations, 100% of NA H5N6 sequences will have a hamming distance of 0 with this primer. **Due to it being a mixed-base, the code reads the mixed-base nucleotides as mismatches. (I am working on fixing this issue and will update this once it is fixed.) Therefore, the actual hamming distance for this primer is 0 and not 1.** 12 H5N6 sequences were used for this alignment and calculations.


## NA H7N9 Forward 1.4 primer NCR hamming distances 

```{r}
NA_H7_F <- read.csv("hamming_distance_results/NA_H7_NA_Forward_1.4_results_NCR.csv")

NA_H7_1 <- NA_H7_F[-c(71, 72),]

barplot(table(NA_H7_1$hamming_distance), main = "NA H7N9 Hamming Distances, forward 1.4 primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,70))

```

**NA Forward 1.4 primer**: *AGC ATC* **AGC AAA AGC AGG** GT

**Melting temperature**: 57.7ºC 

**Hairpins** (ΔG < -1): 1 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 1 (with both reverse primers)


This primer is specific for the NA segment of H5N1 and H7N9 strains of Influenza A. The primer includes a 6 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGC AAA AGC AGG, **bolded**),and part of the non-coding region that is unique to the NA segments. Based on the alignment shade threshold and hamming distance calculations, 73% of NA H7N9 sequences will have a hamming distance of 0 with this primer. 70 H7N9 sequences were used for this alignment and calculations. 


## NA H7N9 Reverse 2.3 primer NCR hamming distances 

```{r}
NA_H7_R <- read.csv("hamming_distance_results/NA_H7_NA_Reverse_2.3_results_NCR.csv")

NA_H7_2 <- NA_H7_R[-c(71, 72),]

barplot(table(NA_H7_2$hamming_distance), main = "NA H7N9 Hamming Distances, reverse 2.3 primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,70))
```


**NA Reverse 2.2 primer**: *GAT GCC* **AGT AGA AAC AAG G**GT *S*T
 
**Reverse compliment (without 5' tail)**: A*G*A C**CC TTG TTT CTA CT**

**Melting temperature**: 57ºC 

**Hairpins** (ΔG < -1): 0 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 1 (with all forward primers)

This primer is specific for the NA segment of H5N6, H7N9, and some H1N1 strains of Influenza A. The reverse compliment of this primer was used for the alignments and hamming distance calculations. This is a mixed base primer, with the mixed base noted in *italics*. The nucleotide that is present in the H9N7 strains is noted in the reverse compliment sequence, also in *italics*. The primer includes a 6 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGT AGA AAC AAG G, **bolded**) and part of the non-coding region that is unique to the NA segments. Based on the alignment shade threshold and hamming distance calculations, 99% of NA H7N9 sequences will have a hamming distance of 0 with this primer. **Due to it being a mixed-base, the code reads the mixed-base nucleotides as mismatches. (I am working on fixing this issue and will update this once it is fixed.) Therefore, the actual hamming distance for this primer is 0 and not 1.** 70 H7N9 sequences were used for this alignment and calculations.


## NA H9N2 Forward 1.3 primer NCR hamming distances 

```{r}
NA_H9_F <- read.csv("hamming_distance_results/NA_H9_NA_Forward_1.3_results_NCR.csv")

NA_H9_1 <- NA_H9_F[-c(15, 16),]

barplot(table(NA_H9_1$hamming_distance), main = "NA H9N2 Hamming Distances, forward 1.3 primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,15))
```

**NA Forward 1.3 primer**: *AGC ATC* **AGC AAA AGC AGG** AGT

**Melting temperature**: 57.3ºC 

**Hairpins** (ΔG < -1): 1 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 1 (with both reverse primers)

This primer is specific for the NA segment of H1N1, H3N2, and H9N2 strains of Influenza A. The primer includes a 6 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGC AAA AGC AGG, **bolded**),and part of the non-coding region that is unique to the NA segments. Based on the alignment shade threshold and hamming distance calculations, 93% of NA H9N2 sequences will have a hamming distance of 0 with this primer. 14 H9N2 sequences were used for this alignment and calculations. 


## NA H9N2 Reverse 2.2 primer NCR hamming distances 

```{r}
NA_H9_R <- read.csv("hamming_distance_results/NA_H9_NA_Reverse_2.2_results_NCR.csv")

NA_H9_2 <- NA_H9_R[-c(15, 16),]

barplot(table(NA_H9_2$hamming_distance), main = "NA H9N2 Hamming Distances, reverse 2.2 primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,15))
```

**NA Reverse 2.2 primer**: *CGA TGC C***AG TAG AAA CAA GG**A GTT 

**Reverse compliment (without 5' tail)**: AAC T**CC TTG TTT CTA CT**

**Melting temperature**: 57.1ºC 

**Hairpins** (ΔG < -1): 0 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 1 (with all forward primers)

This primer is specific for the NA segment of H1N1, H3N2, and H9N2 strains of Influenza A. The reverse compliment of this primer was used for the alignments and hamming distance calculations. The primer includes a 7 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGT AGA AAC AAG G, **bolded**) and part of the non-coding region that is unique to the NA segments. Based on the alignment shade threshold and hamming distance calculations, 100% of NA H9N2 sequences will have a hamming distance of 0 with this primer. 14 H9N2 sequences were used for this alignment and calculations.


In total, **4,819** sequences were used to design the NA primers. These sequences were downloaded from the NCBI Influenza database and the GSAID Influenza database. They were chosen based on length (had to include the non-coding regions) and quality of the sequence. I believe that a **4 primer pool of Forward 1.3 and 1.4 and Reverse 2.2 and 2.3** with all primers in equal concentration would be the best. 