% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  pdftitle={PB1 NCR Hamming Distances Updated},
  pdfauthor={Alexandra Christopoulos},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{PB1 NCR Hamming Distances Updated}
\author{Alexandra Christopoulos}
\date{2023-04-07}

\begin{document}
\maketitle

\hypertarget{pb1-h1n1-forward-1.3-primer-ncr-hamming-distances}{%
\subsection{PB1 H1N1 Forward 1.3 primer NCR hamming
distances}\label{pb1-h1n1-forward-1.3-primer-ncr-hamming-distances}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{PB1\_H1\_F }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"hamming\_distance\_results/PB1\_H1\_PB1\_Forward\_1.3\_results\_NCR.csv"}\NormalTok{)}

\NormalTok{PB1\_H1\_1 }\OtherTok{\textless{}{-}}\NormalTok{ PB1\_H1\_F[}\SpecialCharTok{{-}}\FunctionTok{c}\NormalTok{(}\DecValTok{61}\NormalTok{),]}

\FunctionTok{barplot}\NormalTok{(}\FunctionTok{table}\NormalTok{(PB1\_H1\_1}\SpecialCharTok{$}\NormalTok{hamming\_distance), }\AttributeTok{main =} \StringTok{"PB1 H1N1 hamming distances, forward primer"}\NormalTok{, }\AttributeTok{xlab =} \StringTok{"Hamming Distance"}\NormalTok{, }\AttributeTok{ylab =} \StringTok{"Number of Sequences"}\NormalTok{, }\AttributeTok{ylim =} \FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{60}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\includegraphics{PB1_NCR_hamming_distance_analysis_updated_files/figure-latex/unnamed-chunk-2-1.pdf}

\textbf{PB1 Forward 1.3 primer}: \textbf{AGC AAA AGC AGG} CAA ACC ATT TG

\textbf{Melting temperature}: 57.5ºC

\textbf{Hairpins} (ΔG \textless{} -1): 1

\textbf{Self-dimers} (ΔG \textless{} -6): 0

\textbf{Heterodimers} (ΔG \textless{} -6): 0

This primer is universal for all PB1 segments for all the strains of
Influenza A. The primer consists of the universal non-coding region (AGC
AAA AGC AGG, \textbf{bolded}), and part of the non-coding region that is
unique to the PB1 segments. Based on the alignment shade threshold and
hamming distance calculations, 90\% of PB1 H1N1 sequences will have a
hamming distance of 0 with this primer. 60 H1N1 sequences were used for
this alignment and calculations.

\hypertarget{pb1-h1n1-reverse-2.4-primer-ncr-hamming-distances}{%
\subsection{PB1 H1N1 Reverse 2.4 primer NCR hamming
distances}\label{pb1-h1n1-reverse-2.4-primer-ncr-hamming-distances}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{PB1\_H1\_R }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"hamming\_distance\_results/PB1\_H1\_PB1\_Reverse\_2.4\_results\_NCR.csv"}\NormalTok{)}

\NormalTok{PB1\_H1\_2 }\OtherTok{\textless{}{-}}\NormalTok{ PB1\_H1\_R[}\SpecialCharTok{{-}}\FunctionTok{c}\NormalTok{(}\DecValTok{61}\NormalTok{),]}

\FunctionTok{barplot}\NormalTok{(}\FunctionTok{table}\NormalTok{(PB1\_H1\_2}\SpecialCharTok{$}\NormalTok{hamming\_distance), }\AttributeTok{main =} \StringTok{"PB1 H1N1 hamming distances, reverse primer"}\NormalTok{, }\AttributeTok{xlab =} \StringTok{"Hamming Distance"}\NormalTok{, }\AttributeTok{ylab =} \StringTok{"Number of Sequences"}\NormalTok{, }\AttributeTok{ylim =} \FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{60}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\includegraphics{PB1_NCR_hamming_distance_analysis_updated_files/figure-latex/unnamed-chunk-3-1.pdf}

\textbf{PB1 Reverse 2.4 primer}: \emph{CGA T}\textbf{AG TAG AAA CAA GG}C
ATT TTT TCA TGA AGG

\textbf{Reverse compliment (without 5' tail)}: CCT TCA TGA AAA AAT
G\textbf{CC TTG TTT CTA CT}

\textbf{Melting temperature}: 58.1ºC

\textbf{Hairpins} (ΔG \textless{} -1): 0

\textbf{Self-dimers} (ΔG \textless{} -6): 1

\textbf{Heterodimers} (ΔG \textless{} -6): 0

This primer is universal for all PB1 segments of all strains of
Influenza A. The reverse compliment of this primer was used for the
alignments and hamming distance calculations. The primer includes a 4
nucleotide long 5' tail (\emph{italized}) that is not based on the
sequences in order to increase the melting temperature and improve other
factors of the primer; this tail was not included in the hamming
distance calculations. The rest of the primer consists of the universal
non-coding region (AGT AGA AAC AAG G, \textbf{bolded}) and part of the
non-coding region that is unique to the PB1 segments. Based on the
alignment shade threshold and hamming distance calculations, 98\% of PB1
H1N1 sequences will have a hamming distance of 0 with this primer. 60
H1N1 sequences were used for this alignment and calculations.

\hypertarget{pb1-h3nx-forward-1.3-primer-ncr-hamming-distances}{%
\subsection{PB1 H3Nx Forward 1.3 primer NCR hamming
distances}\label{pb1-h3nx-forward-1.3-primer-ncr-hamming-distances}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{PB1\_H3\_F }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"hamming\_distance\_results/PB1\_H3\_PB1\_Forward\_1.3\_results\_NCR.csv"}\NormalTok{)}

\NormalTok{PB1\_H3\_1 }\OtherTok{\textless{}{-}}\NormalTok{ PB1\_H3\_F[}\SpecialCharTok{{-}}\FunctionTok{c}\NormalTok{(}\DecValTok{3673}\NormalTok{),]}

\FunctionTok{barplot}\NormalTok{(}\FunctionTok{table}\NormalTok{(PB1\_H3\_1}\SpecialCharTok{$}\NormalTok{hamming\_distance), }\AttributeTok{main =} \StringTok{"PB1 H3Nx hamming distances, forward primer"}\NormalTok{, }\AttributeTok{xlab =} \StringTok{"Hamming Distance"}\NormalTok{, }\AttributeTok{ylab =} \StringTok{"Number of Sequences"}\NormalTok{, }\AttributeTok{ylim =} \FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{3500}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\includegraphics{PB1_NCR_hamming_distance_analysis_updated_files/figure-latex/unnamed-chunk-4-1.pdf}

\textbf{PB1 Forward 1.3 primer}: \textbf{AGC AAA AGC AGG} CAA ACC ATT TG

\textbf{Melting temperature}: 57.5ºC

\textbf{Hairpins} (ΔG \textless{} -1): 1

\textbf{Self-dimers} (ΔG \textless{} -6): 0

\textbf{Heterodimers} (ΔG \textless{} -6): 0

This primer is universal for all PB1 segments for all the strains of
Influenza A. The primer consists of the universal non-coding region (AGC
AAA AGC AGG, \textbf{bolded}), and part of the non-coding region that is
unique to the PB1 segments. Based on the alignment shade threshold and
hamming distance calculations, 80\% of PB1 H3Nx sequences will have a
hamming distance of 0 with this primer. 3,671 H3N2 sequences and 1 H3N8
sequence were used for this alignment and calculations.

\hypertarget{pb1-h3nx-reverse-2.4-ncr-hamming-distance-results}{%
\subsection{PB1 H3Nx Reverse 2.4 NCR hamming distance
results}\label{pb1-h3nx-reverse-2.4-ncr-hamming-distance-results}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{PB1\_H3\_R }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"hamming\_distance\_results/PB1\_H3\_PB1\_Reverse\_2.4\_results\_NCR.csv"}\NormalTok{)}

\NormalTok{PB1\_H3\_2 }\OtherTok{\textless{}{-}}\NormalTok{ PB1\_H3\_R[}\SpecialCharTok{{-}}\FunctionTok{c}\NormalTok{(}\DecValTok{3673}\NormalTok{),]}

\FunctionTok{barplot}\NormalTok{(}\FunctionTok{table}\NormalTok{(PB1\_H3\_2}\SpecialCharTok{$}\NormalTok{hamming\_distance), }\AttributeTok{main =} \StringTok{"PB1 H3Nx hamming distances, reverse primer"}\NormalTok{, }\AttributeTok{xlab =} \StringTok{"Hamming Distance"}\NormalTok{, }\AttributeTok{ylab =} \StringTok{"Number of Sequences"}\NormalTok{, }\AttributeTok{ylim =} \FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{4000}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\includegraphics{PB1_NCR_hamming_distance_analysis_updated_files/figure-latex/unnamed-chunk-5-1.pdf}

\textbf{PB1 Reverse 2.4 primer}: \emph{CGA T}\textbf{AG TAG AAA CAA GG}C
ATT TTT TCA TGA AGG

\textbf{Reverse compliment (without 5' tail)}: CCT TCA TGA AAA AAT
G\textbf{CC TTG TTT CTA CT}

\textbf{Melting temperature}: 58.1ºC

\textbf{Hairpins} (ΔG \textless{} -1): 0

\textbf{Self-dimers} (ΔG \textless{} -6): 1

\textbf{Heterodimers} (ΔG \textless{} -6): 0

This primer is universal for all PB1 segments of all strains of
Influenza A. The reverse compliment of this primer was used for the
alignments and hamming distance calculations. The primer includes a 4
nucleotide long 5' tail (\emph{italized}) that is not based on the
sequences in order to increase the melting temperature and improve other
factors of the primer; this tail was not included in the hamming
distance calculations. The rest of the primer consists of the universal
non-coding region (AGT AGA AAC AAG G, \textbf{bolded}) and part of the
non-coding region that is unique to the PB1 segments. Based on the
alignment shade threshold and hamming distance calculations, 99\% of PB1
H3Nx sequences will have a hamming distance of 0 with this primer. 3,671
H3N2 sequences and 1 H3N8 sequence were used for this alignment and
calculations.

\hypertarget{pb1-h5nx-forward-1.3-primer-ncr-hamming-distances}{%
\subsection{PB1 H5Nx Forward 1.3 primer NCR hamming
distances}\label{pb1-h5nx-forward-1.3-primer-ncr-hamming-distances}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{PB1\_H5\_F }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"hamming\_distance\_results/PB1\_H5\_PB1\_Forward\_1.3\_results\_NCR.csv"}\NormalTok{)}

\NormalTok{PB1\_H5\_1 }\OtherTok{\textless{}{-}}\NormalTok{ PB1\_H5\_F[}\SpecialCharTok{{-}}\FunctionTok{c}\NormalTok{(}\DecValTok{8}\NormalTok{),]}

\FunctionTok{barplot}\NormalTok{(}\FunctionTok{table}\NormalTok{(PB1\_H5\_1}\SpecialCharTok{$}\NormalTok{hamming\_distance), }\AttributeTok{main =} \StringTok{"PB1 H5Nx hamming distances, forward primer"}\NormalTok{, }\AttributeTok{xlab =} \StringTok{"Hamming Distance"}\NormalTok{, }\AttributeTok{ylab =} \StringTok{"Number of Sequences"}\NormalTok{, }\AttributeTok{ylim =} \FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{8}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\includegraphics{PB1_NCR_hamming_distance_analysis_updated_files/figure-latex/unnamed-chunk-6-1.pdf}

\textbf{PB1 Forward 1.3 primer}: \textbf{AGC AAA AGC AGG} CAA ACC ATT TG

\textbf{Melting temperature}: 57.5ºC

\textbf{Hairpins} (ΔG \textless{} -1): 1

\textbf{Self-dimers} (ΔG \textless{} -6): 0

\textbf{Heterodimers} (ΔG \textless{} -6): 0

This primer is universal for all PB1 segments for all the strains of
Influenza A. The primer consists of the universal non-coding region (AGC
AAA AGC AGG, \textbf{bolded}), and part of the non-coding region that is
unique to the PB1 segments. Based on the alignment shade threshold and
hamming distance calculations, 75\% of PB1 H5Nx sequences will have a
hamming distance of 0 with this primer. 5 H5N6 sequences, 1 H5N2
sequence, and 1 H5N8 (7 in total) were used for this alignment and
calculations.

\hypertarget{pb1-h5nx-reverse-2.4-primer-ncr-hamming-distances}{%
\subsection{PB1 H5Nx Reverse 2.4 primer NCR hamming
distances}\label{pb1-h5nx-reverse-2.4-primer-ncr-hamming-distances}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{PB1\_H5\_R }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"hamming\_distance\_results/PB1\_H5\_PB1\_Reverse\_2.4\_results\_NCR.csv"}\NormalTok{)}

\NormalTok{PB1\_H5\_2 }\OtherTok{\textless{}{-}}\NormalTok{ PB1\_H5\_R[}\SpecialCharTok{{-}}\FunctionTok{c}\NormalTok{(}\DecValTok{8}\NormalTok{),]}

\FunctionTok{barplot}\NormalTok{(}\FunctionTok{table}\NormalTok{(PB1\_H5\_2}\SpecialCharTok{$}\NormalTok{hamming\_distance), }\AttributeTok{main =} \StringTok{"PB1 H5Nx hamming distances, reverse primer"}\NormalTok{, }\AttributeTok{xlab =} \StringTok{"Hamming Distance"}\NormalTok{, }\AttributeTok{ylab =} \StringTok{"Number of Sequences"}\NormalTok{, }\AttributeTok{ylim =} \FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{8}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\includegraphics{PB1_NCR_hamming_distance_analysis_updated_files/figure-latex/unnamed-chunk-7-1.pdf}

\textbf{PB1 Reverse 2.4 primer}: \emph{CGA T}\textbf{AG TAG AAA CAA GG}C
ATT TTT TCA TGA AGG

\textbf{Reverse compliment (without 5' tail)}: CCT TCA TGA AAA AAT
G\textbf{CC TTG TTT CTA CT}

\textbf{Melting temperature}: 58.1ºC

\textbf{Hairpins} (ΔG \textless{} -1): 0

\textbf{Self-dimers} (ΔG \textless{} -6): 1

\textbf{Heterodimers} (ΔG \textless{} -6): 0

This primer is universal for all PB1 segments of all strains of
Influenza A. The reverse compliment of this primer was used for the
alignments and hamming distance calculations. The primer includes a 4
nucleotide long 5' tail (\emph{italized}) that is not based on the
sequences in order to increase the melting temperature and improve other
factors of the primer; this tail was not included in the hamming
distance calculations. The rest of the primer consists of the universal
non-coding region (AGT AGA AAC AAG G, \textbf{bolded}) and part of the
non-coding region that is unique to the PB1 segments. Based on the
alignment shade threshold and hamming distance calculations, 100\% of
PB1 H5Nx sequences will have a hamming distance of 0 with this primer. 5
H5N6 sequences, 1 H5N2 sequence, and 1 H5N8 sequence (7 in total) were
used for this alignment and calculations.

\hypertarget{pb1-h7n9-forward-1.3-primer-ncr-hamming-distances}{%
\subsection{PB1 H7N9 Forward 1.3 primer NCR hamming
distances}\label{pb1-h7n9-forward-1.3-primer-ncr-hamming-distances}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{PB1\_H7\_F }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"hamming\_distance\_results/PB1\_H7\_PB1\_Forward\_1.3\_results\_NCR.csv"}\NormalTok{)}

\NormalTok{PB1\_H7\_1 }\OtherTok{\textless{}{-}}\NormalTok{ PB1\_H7\_F[}\SpecialCharTok{{-}}\FunctionTok{c}\NormalTok{(}\DecValTok{54}\NormalTok{),]}

\FunctionTok{barplot}\NormalTok{(}\FunctionTok{table}\NormalTok{(PB1\_H7\_1}\SpecialCharTok{$}\NormalTok{hamming\_distance), }\AttributeTok{main =} \StringTok{"PB1 H7N9 hamming distances, forward primer"}\NormalTok{, }\AttributeTok{xlab =} \StringTok{"Hamming Distance"}\NormalTok{, }\AttributeTok{ylab =} \StringTok{"Number of Sequences"}\NormalTok{, }\AttributeTok{ylim =} \FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{30}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\includegraphics{PB1_NCR_hamming_distance_analysis_updated_files/figure-latex/unnamed-chunk-8-1.pdf}

\textbf{PB1 Forward 1.3 primer}: \textbf{AGC AAA AGC AGG} CAA ACC ATT TG

\textbf{Melting temperature}: 57.5ºC

\textbf{Hairpins} (ΔG \textless{} -1): 1

\textbf{Self-dimers} (ΔG \textless{} -6): 0

\textbf{Heterodimers} (ΔG \textless{} -6): 0

This primer is universal for all PB1 segments for all the strains of
Influenza A. The primer consists of the universal non-coding region (AGC
AAA AGC AGG, \textbf{bolded}), and part of the non-coding region that is
unique to the PB1 segments. Based on the alignment shade threshold and
hamming distance calculations, 55\% of PB1 H7N9 sequences will have a
hamming distance of 0 with this primer. 53 H7N9 sequences were used for
this alignment and calculations.

\hypertarget{pb1-h7n9-reverse-2.4-primer-ncr-hamming-distances}{%
\subsection{PB1 H7N9 Reverse 2.4 primer NCR hamming
distances}\label{pb1-h7n9-reverse-2.4-primer-ncr-hamming-distances}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{PB1\_H7\_R }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"hamming\_distance\_results/PB1\_H7\_PB1\_Reverse\_2.4\_results\_NCR.csv"}\NormalTok{)}

\NormalTok{PB1\_H7\_2 }\OtherTok{\textless{}{-}}\NormalTok{ PB1\_H7\_R[}\SpecialCharTok{{-}}\FunctionTok{c}\NormalTok{(}\DecValTok{54}\NormalTok{),]}

\FunctionTok{barplot}\NormalTok{(}\FunctionTok{table}\NormalTok{(PB1\_H7\_2}\SpecialCharTok{$}\NormalTok{hamming\_distance), }\AttributeTok{main =} \StringTok{"PB1 H7N9 hamming distances, forward primer"}\NormalTok{, }\AttributeTok{xlab =} \StringTok{"Hamming Distance"}\NormalTok{, }\AttributeTok{ylab =} \StringTok{"Number of Sequences"}\NormalTok{, }\AttributeTok{ylim =} \FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{60}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\includegraphics{PB1_NCR_hamming_distance_analysis_updated_files/figure-latex/unnamed-chunk-9-1.pdf}

\textbf{PB1 Reverse 2.4 primer}: \emph{CGA T}\textbf{AG TAG AAA CAA GG}C
ATT TTT TCA TGA AGG

\textbf{Reverse compliment (without 5' tail)}: CCT TCA TGA AAA AAT
G\textbf{CC TTG TTT CTA CT}

\textbf{Melting temperature}: 58.1ºC

\textbf{Hairpins} (ΔG \textless{} -1): 0

\textbf{Self-dimers} (ΔG \textless{} -6): 1

\textbf{Heterodimers} (ΔG \textless{} -6): 0

This primer is universal for all PB1 segments of all strains of
Influenza A. The reverse compliment of this primer was used for the
alignments and hamming distance calculations. The primer includes a 4
nucleotide long 5' tail (\emph{italized}) that is not based on the
sequences in order to increase the melting temperature and improve other
factors of the primer; this tail was not included in the hamming
distance calculations. The rest of the primer consists of the universal
non-coding region (AGT AGA AAC AAG G, \textbf{bolded}) and part of the
non-coding region that is unique to the PB1 segments. Based on the
alignment shade threshold and hamming distance calculations, 98\% of PB1
H7N9 sequences will have a hamming distance of 0 with this primer. 53
H7N9 sequences were used for this alignment and calculations.

\hypertarget{pb1-h9n2-forward-1.3-primer-ncr-hamming-distances}{%
\subsection{PB1 H9N2 Forward 1.3 primer NCR hamming
distances}\label{pb1-h9n2-forward-1.3-primer-ncr-hamming-distances}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{PB1\_H9\_F }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"hamming\_distance\_results/PB1\_H9\_PB1\_Forward\_1.3\_results\_NCR.csv"}\NormalTok{)}

\NormalTok{PB1\_H9\_1 }\OtherTok{\textless{}{-}}\NormalTok{ PB1\_H9\_F[}\SpecialCharTok{{-}}\FunctionTok{c}\NormalTok{(}\DecValTok{16}\NormalTok{),]}

\FunctionTok{barplot}\NormalTok{(}\FunctionTok{table}\NormalTok{(PB1\_H9\_1}\SpecialCharTok{$}\NormalTok{hamming\_distance), }\AttributeTok{main =} \StringTok{"PB1 H9N2 hamming distances, forward primer"}\NormalTok{, }\AttributeTok{xlab =} \StringTok{"Hamming Distance"}\NormalTok{, }\AttributeTok{ylab =} \StringTok{"Number of Sequences"}\NormalTok{, }\AttributeTok{ylim =} \FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{15}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\includegraphics{PB1_NCR_hamming_distance_analysis_updated_files/figure-latex/unnamed-chunk-10-1.pdf}

\textbf{PB1 Forward 1.3 primer}: \textbf{AGC AAA AGC AGG} CAA ACC ATT TG

\textbf{Melting temperature}: 57.5ºC

\textbf{Hairpins} (ΔG \textless{} -1): 1

\textbf{Self-dimers} (ΔG \textless{} -6): 0

\textbf{Heterodimers} (ΔG \textless{} -6): 0

This primer is universal for all PB1 segments for all the strains of
Influenza A. The primer consists of the universal non-coding region (AGC
AAA AGC AGG, \textbf{bolded}), and part of the non-coding region that is
unique to the PB1 segments. Based on the alignment shade threshold and
hamming distance calculations, 87\% of PB1 H9N2 sequences will have a
hamming distance of 0 with this primer. 15 H9N2 sequences were used for
this alignment and calculations.

\hypertarget{pb1-h9n2-reverse-2.4-primer-ncr-hamming-distances}{%
\subsection{PB1 H9N2 Reverse 2.4 primer NCR hamming
distances}\label{pb1-h9n2-reverse-2.4-primer-ncr-hamming-distances}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{PB1\_H9\_R }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"hamming\_distance\_results/PB1\_H9\_PB1\_Reverse\_2.4\_results\_NCR.csv"}\NormalTok{)}

\NormalTok{PB1\_H9\_2 }\OtherTok{\textless{}{-}}\NormalTok{ PB1\_H9\_R[}\SpecialCharTok{{-}}\FunctionTok{c}\NormalTok{(}\DecValTok{16}\NormalTok{),]}

\FunctionTok{barplot}\NormalTok{(}\FunctionTok{table}\NormalTok{(PB1\_H9\_2}\SpecialCharTok{$}\NormalTok{hamming\_distance), }\AttributeTok{main =} \StringTok{"PB1 H9N2 hamming distances, reverse primer"}\NormalTok{, }\AttributeTok{xlab =} \StringTok{"Hamming Distance"}\NormalTok{, }\AttributeTok{ylab =} \StringTok{"Number of Sequences"}\NormalTok{, }\AttributeTok{ylim =} \FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{15}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\includegraphics{PB1_NCR_hamming_distance_analysis_updated_files/figure-latex/unnamed-chunk-11-1.pdf}

\textbf{PB1 Reverse 2.4 primer}: \emph{CGA T}\textbf{AG TAG AAA CAA GG}C
ATT TTT TCA TGA AGG

\textbf{Reverse compliment (without 5' tail)}: CCT TCA TGA AAA AAT
G\textbf{CC TTG TTT CTA CT}

\textbf{Melting temperature}: 58.1ºC

\textbf{Hairpins} (ΔG \textless{} -1): 0

\textbf{Self-dimers} (ΔG \textless{} -6): 1

\textbf{Heterodimers} (ΔG \textless{} -6): 0

This primer is universal for all PB1 segments of all strains of
Influenza A. The reverse compliment of this primer was used for the
alignments and hamming distance calculations. The primer includes a 4
nucleotide long 5' tail (\emph{italized}) that is not based on the
sequences in order to increase the melting temperature and improve other
factors of the primer; this tail was not included in the hamming
distance calculations. The rest of the primer consists of the universal
non-coding region (AGT AGA AAC AAG G, \textbf{bolded}) and part of the
non-coding region that is unique to the PB1 segments. Based on the
alignment shade threshold and hamming distance calculations, 87\% of PB1
H9N2 sequences will have a hamming distance of 0 with this primer. 15
H9N2 sequences were used for this alignment and calculations.

In total, \textbf{3,807 sequences} were used to design the PB1 primers.
These sequences were downloaded from the NCBI Influenza database and the
GSAID Influenza database. They were chosen based on length (had to
include the non-coding regions) and quality of the sequence.

\end{document}
