---
title: "HA NCR Hamming Distances Updated"
author: "Alexandra Christopoulos"
date: "2023-04-04"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(root.dir = "C:/Users/Lexie Christopoulos/Repos/GenomeSequencing-/db/", echo = TRUE)
```


```{r, include=FALSE}
#in console: setwd("C:/Users/Lexie Christopoulos/Repos/GenomeSequencing-/db/HA")
```

## HA H1N1 Forward 1.13 primer NCR hamming distances 

```{r}
HA_H1_F1 <-read.csv("hamming_distance_results/HA_H1_HA_Forward_1.13_results_NCR.csv")

HA_H1_1 <- HA_H1_F1[-c(3403),]

barplot(table(HA_H1_1$hamming_distance),main = "HA H1N1 Hamming Distances, forward primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,3500))

```

**HA Forward 1.13 primer**: *AGC AT***A GCA AAA GCA GG**G GA 

**Melting temperature**: 57ºC 

**Hairpins** (ΔG < -1): 0 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 0

This primer is universal for all HA segments of all strains of Influenza A. It includes a 5 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGC AAA AGC AGG, **bolded**) and part of the non-coding region that is unique to the HA segments. Based on the alignment shade threshold and hamming distance calculations, 88% of HA H1N1 sequences will have a hamming distance of 0 with this primer  with this primer. 3,402 sequences were used for this alignment and calculations. 


## HA H1N1 Reverse 2.7 primer NCR hamming distances 

```{r}
HA_H1_R <-read.csv("hamming_distance_results/HA_H1_HA_Reverse_2.7_results_NCR.csv")

HA_H1_3 <- HA_H1_R[-c(3403),]

barplot(table(HA_H1_3$hamming_distance),main = "HA H1N1 Hamming Distances, reverse primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,4000))
```

**HA Reverse 2.7 primer**: *GCG ACT* **AGT AGA AAC AAG G**GT GTT TTT

**Melting temperature**: 57.4ºC 

**Hairpins** (ΔG < -1): 3

**Self-dimers** (ΔG < -6): 2

**Heterodimers** (ΔG < -6): 0 

This primer is universal for all HA segments of all strains of Influenza A. The reverse compliment of this primer was used for the alignments and hamming distance calculations. The primer includes a 6 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGT AGA AAC AAG G, **bolded**) and part of the non-coding region that is unique to the HA segments. Based on the alignment shade threshold and hamming distance calculations, 99% of H1N1 sequences will have a hamming distance of 0 with this primer  with this primer. 3,402 sequences were used for this alignment and calculation. 


## HA H3NX Forward 1.13 primer NCR hamming distances 

```{r}
HA_H3_F1 <-read.csv("hamming_distance_results/HA_H3_HA_Forward_1.13_results_NCR.csv")
HA_H3_F2 <-read.csv("hamming_distance_results/HA_H3_HA_Forward_1.13_results_NCR_H3N8.csv")

HA_H3_1 <- HA_H3_F1[-c(5261),]
HA_H3_2 <- HA_H3_F2[-c(8),]

barplot(table(HA_H3_1$hamming_distance), main = "HA H3N2 Hamming Distances ,forward primer ", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,5500))
barplot(table(HA_H3_2$hamming_distance), main = "HA H3N8 Hamming Distances, forward primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,10))
```

**HA Forward 1.13 primer**: *AGC AT***A GCA AAA GCA GG**G GA

**Melting temperature**: 57ºC 

**Hairpins** (ΔG < -1): 0 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 0

This primer is universal for all HA segments of all strains of Influenza A. It includes a 5 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGC AAA AGC AGG, **bolded**) and part of the non-coding region that is unique to the HA segments. Based on the alignment shade threshold and hamming distance calculations, 91% of HA H3Nx sequences will have a hamming distance of 0 with this primer . (91% of H3N2 sequences and 100% of H3N8 sequences). 5,260 H3N2 sequences and 7 H3N8 sequences were used for these alignments and calculations. 


## HA H3Nx Reverse 2.7 primer NCR hamming distances 

```{r}
HA_H3_R1 <-read.csv("hamming_distance_results/HA_H3_HA_Reverse_2.7_results_NCR.csv")
HA_H3_R2 <-read.csv("hamming_distance_results/HA_H3_HA_Reverse_2.7_results_NCR_H3N8.csv")

HA_H3_3 <- HA_H3_R1[-c(5261),]
HA_H3_4 <- HA_H3_R2[-c(8),]

barplot(table(HA_H3_3$hamming_distance), main = "HA H3N2 Hamming Distances, reverse primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,6000))
barplot(table(HA_H3_4$hamming_distance), main = "HA H3N8 Hamming Distances , reverse primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,10))

```

**HA Reverse 2.7 primer**: *GCG ACT* **AGT AGA AAC AAG G**GT GTT TTT

**Melting temperature**: 57.4ºC

**Hairpins** (ΔG < -1): 3

**Self-dimers** (ΔG < -6): 2

**Heterodimers** (ΔG < -6): 0 

This primer is universal for all HA segments of all strains of Influenza A. The reverse compliment of this primer was used for the alignments and hamming distance calculations. The primer includes a 6 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGT AGA AAC AAG G, **bolded**) and part of the non-coding region that is unique to the HA segments (bolded). Based on the alignment shade threshold and hamming distance calculations, 99% of H3Nx sequences will have a hamming distance of 0 with this primer . (99% of H3N2 sequences, and 100% of H3N8 sequences). 5,260 H3N2 sequences and 7 H3N8 sequences were used for these alignments and calculations. 



## HA H5Nx Forward 1.13 primer NCR hamming distances 

```{r}
HA_H5_F1 <-read.csv("hamming_distance_results/HA_H5_HA_Forward_1.13_results_NCR.csv")

HA_H5_1 <- HA_H5_F1[-c(15),]

barplot(table(HA_H5_1$hamming_distance), main = "HA H5Nx Hamming Distances, forward primer", xlab = "Hamming Distance", ylab = "Number of Sequences",ylim = c(0,15))
```

**HA Forward 1.13 primer**: *AGC AT***A GCA AAA GCA GG**G GA

**Melting temperature**: 57ºC

**Hairpins** (ΔG < -1): 0 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 0

This primer is universal for all HA segments of all strains of Influenza A. It includes a 5 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGC AAA AGC AGG, **bolded**) and part of the non-coding region that is unique to the HA segments. Based on the alignment shade threshold and hamming distance calculations, 80% of HA H5Nx sequences will have a hamming distance of 1 with this primer. (For HA H5Nx sequences, the last A is actually a T, and therefore there will always be 1 mismatch.) 11 H5N6, 2 H5N1, and 1 H5N8 (13 in total) sequences were used for this alignment and calculations. 


## HA H5Nx 2.7 Reverse primer NCR hamming distances 

```{r}
HA_H5_R <-read.csv("hamming_distance_results/HA_H5_HA_Reverse_2.7_results_NCR.csv")

HA_H5_3 <- HA_H5_R[-c(15),]

barplot(table(HA_H5_3$hamming_distance), main = "HA H5Nx Hamming Distances, reverse primer", xlab = "Hamming Distance", ylab = "Number of Sequences", ylim = c(0,15))
```

**HA Reverse 2.7 primer**: *GCG ACT* **AGT AGA AAC AAG G**GT GTT TTT

**Melting temperature**: 57.4ºC 

**Hairpins** (ΔG < -1): 3

**Self-dimers** (ΔG < -6): 2

**Heterodimers** (ΔG < -6): 0 

This primer is universal for all HA segments of all strains of Influenza A. The reverse compliment of this primer was used for the alignments and hamming distance calculations. The primer includes a 6 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGT AGA AAC AAG G, **bolded**) and part of the non-coding region that is unique to the HA segments. Based on the alignment shade threshold and hamming distance calculations, 100% of H5Nx sequences will have a hamming distance of 0 with this primer. 11 H5N6, 2 H5N1, and 1 H5N8 (13 in total) sequences were used for this alignment and calculations. 



## HA H7N9 Forward 1.13 primer NCR hamming distances 

```{r}
HA_H7_F1 <-read.csv("hamming_distance_results/HA_H7_HA_Forward_1.13_results_NCR.csv")

HA_H7_1 <- HA_H7_F1[-c(58),]

barplot(table(HA_H7_1$hamming_distance), main= "HA H7N9 Hamming Distances, forward primer", xlab = "Hamming Distance", ylab = "Number of Sequences",ylim = c(0,60))

```

**HA Forward 1.13 primer**: *AGC AT***A GCA AAA GCA GG**G GA

**Melting temperature**: 57ºC 

**Hairpins** (ΔG < -1): 0 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 0

This primer is universal for all HA segments of all strains of Influenza A. It includes a 5 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGC AAA AGC AGG, **bolded**) and part of the non-coding region that is unique to the HA segments. Based on the alignment shaded threshold and hamming distance calculations, 70% of HA H7N9 sequences will have a hamming distance of 0 with this primer. 57 H7N9 sequences were used for this alignment and calculations. 


## HA H7N9 Reverse 2.7 primer NCR hamming distances 

```{r}
HA_H7_R <-read.csv("hamming_distance_results/HA_H7_HA_Reverse_2.7_results_NCR.csv")

HA_H7_3 <- HA_H7_R[-c(58),]

barplot(table(HA_H7_3$hamming_distance), main = "HA H7N9 Hamming Distances, reverse primer",  xlab = "Hamming Distance", ylab = "Number of Sequences",ylim = c(0,60))

```

**HA Reverse 2.7 primer**: *GCG ACT* **AGT AGA AAC AAG G**GT GTT TTT

**Melting temperature**: 57.4ºC  

**Hairpins** (ΔG < -1): 3

**Self-dimers** (ΔG < -6): 2

**Heterodimers** (ΔG < -6): 0 

This primer is universal for all HA segments of all strains of Influenza A. The reverse compliment of this primer was used for the alignments and hamming distance calculations. The primer includes a 6 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGT AGA AAC AAG G, **bolded**) and part of the non-coding region that is unique to the HA segments. Based on the alignment shade threshold and hamming distance calculations, 100% of H7N9 sequences will have a hamming distance of 0 with this primer. 57 H7N9 sequences were used for this  alignment and calculations. 


## HA H9N2 Forward 1.13 primer NCR hamming distances 

```{r}
HA_H9_F1 <-read.csv("hamming_distance_results/HA_H9_HA_Forward_1.13_results_NCR.csv")

HA_H9_1 <- HA_H9_F1[-c(21),]

barplot(table(HA_H9_1$hamming_distance), main = "HA H9N2 Hamming Distances, forward primer", xlab = "Hamming Distance", ylab = "Number of Sequences",ylim = c(0,20))
```

**HA Forward 1.13 primer**: *AGC AT***A GCA AAA GCA GG**G GA

**Melting temperature**: 57ºC 

**Hairpins** (ΔG < -1): 0 

**Self-dimers** (ΔG < -6): 0

**Heterodimers** (ΔG < -6): 0

This primer is universal for all HA segments of all strains of Influenza A. It includes a 5 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGC AAA AGC AGG, **bolded**) and part of the non-coding region that is unique to the HA segments. Based on the alignment shade threshold and hamming distance calculations, 90% of HA H9N2 sequences will have a hamming distance of 0 with this primer. 20 H9N2 sequences were used for this alignment and calucaltions. 


## HA H9N2 Reverse 2.7 primer NCR hamming distances 

```{r}
HA_H9_R <-read.csv("hamming_distance_results/HA_H9_HA_Reverse_2.7_results_NCR.csv")

HA_H9_3 <- HA_H9_R[-c(21),]

barplot(table(HA_H9_3$hamming_distance), main = "HA H9N2 Hamming Distances, reverse primer", xlab = "Hamming Distance", ylab = "Number of Sequences",ylim = c(0,20))

```

**HA Reverse 2.7 primer**: *GCG ACT* **AGT AGA AAC AAG G**GT GTT TTT

**Melting temperature**: 57.4ºC

**Hairpins** (ΔG < -1): 3

**Self-dimers** (ΔG < -6): 2

**Heterodimers** (ΔG < -6): 0 

This primer is universal for all HA segments of all strains of Influenza A. The reverse compliment of this primer was used for the alignments and hamming distance calculations. The primer includes a 6 nucleotide long 5' tail (*italized*) that is not based on the sequences in order to increase the melting temperature and improve other factors of the primer; this tail was not included in the hamming distance calculations. The rest of the primer consists of the universal non-coding region (AGT AGA AAC AAG G, **bolded**) and part of the non-coding region that is unique to the HA segments. Based on the alignment shade threshold and hamming distance calculations, 100% of H9N2 sequences will have a hamming distance of 0 with this primer. 20 H9N2 sequences were used for this alignment and calucaltions. 


In total, **8,760 sequences** were used to design the HA primers. 
